state-sync/state-sync-v1/src/executor_proxy.rs

```diff
@@ -260,7 +325,7 @@ impl ExecutorProxyTrait for ExecutorProxy {
             .collect::<HashSet<_>>();
 
         // calculate deltas
-        let new_configs = Self::fetch_all_configs(&*self.storage)?;
+        let new_configs = Self::fetch_all_configs(ON_CHAIN_CONFIG_REGISTRY, &*self.storage)?;
```

Comment:
we need to change line 334 as well. otherwise it'd panic if a new config is published.

Reply:
Aah, thanks @zekun000! The reason I missed this is because (in my mind) all of this functionality is going to be removed once the new EventSubscriptionService lands (https://github.com/diem/diem/pull/9117). I think calculating these "config deltas" is unnecessarily complicated and confusing. Instead, listeners should be notified about all reconfiguration events and handle them appropriately (changes, or not). IIUC, this is how all listeners already work today. If we ever need this complicated diffing feature, we can adopt it in the future. For now, I've patched this, but it will be removed soon 😄